src/baal/calibration/calibration.py

```diff
@@ -0,0 +1,124 @@
+import structlog
+from copy import deepcopy
+import torch
+from torch.optim import Adam
+from torch import nn
+from baal.utils.metrics import ECE
+
+log = structlog.get_logger("Calibrating...")
+
+
+class DirichletCalibrator(object):
+    """
+    Adding a linear layer to a classifier model after the model is
+    trained and train this new layer until convergence.
+    Together with the linear layer, the model is now calibrated.
+    Source: https://arxiv.org/abs/1910.12656
+
+    Args:
+
+        wrapper (baal.ModelWrapper): Provides training and testing methods.
+        num_classes (int): Number of classes in classification task.
+        lr (float): Learning rate.
+        reg_factor (float): Regularization factor for the linear layer weights.
+        mu (float): Regularization factor for the linear layer biases.
+            If not given, will be initialized by "l".
+
+    """
+    def __init__(self, wrapper, num_classes, lr, reg_factor, mu=None):
+        self.wrapper = wrapper
+        self.init_model = deepcopy(wrapper.model)
+        self.num_classes = num_classes
+        self.loss = nn.CrossEntropyLoss()
+        self.lr = lr
+        self.reg_factor = reg_factor
+        self.mu = mu or reg_factor
+
+        # TODO: need to support kwargs for initializer for ece_per_class
+        self.wrapper.add_metric("ece", lambda: ECE())
```

Comment:
```suggestion
        self.wrapper.add_metric("ece", lambda: ECE(num_classes))
```

This should work @parmidaatg ?